sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -63,6 +65,27 @@ def decode(self, encoded):
     """Encodes an object to an unnested string."""
     raise NotImplementedError
 
+  def estimate_size(self, value):
+    """Estimates the encoded size of the given value, in bytes."""
+    return len(self.encode(value))
```

Comment:
The default should use ByteCountingOutputStream, or at least override this in StreamCoderImpl to do so.